src/Middleware/HttpLogging/test/HttpLoggingMiddlewareTests.cs

```diff
@@ -0,0 +1,757 @@
+            Assert.DoesNotContain(TestSink.Writes, w => w.Message.Contains("QueryString: ?foo"));
+            Assert.DoesNotContain(TestSink.Writes, w => w.Message.Contains("Connection: keep-alive"));
+            Assert.DoesNotContain(TestSink.Writes, w => w.Message.Contains("Body: test"));
+            Assert.DoesNotContain(TestSink.Writes, w => w.Message.Contains("StatusCode: 200"));
+        }
+
+        [Fact]
+        public async Task DefaultRequestInfoOnlyHeadersAndRequestInfo()
+        {
+            var middleware = new HttpLoggingMiddleware(
+                c =>
+                {
+                    return Task.CompletedTask;
+                },
+                CreateOptionsAccessor(),
+                LoggerFactory);
+
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Protocol = "HTTP/1.0";
+            httpContext.Request.Method = "GET";
+            httpContext.Request.Scheme = "http";
+            httpContext.Request.Path = new PathString("/foo");
+            httpContext.Request.PathBase = new PathString("/foo");
+            httpContext.Request.QueryString = new QueryString("?foo");
+            httpContext.Request.Headers["Connection"] = "keep-alive";
+            httpContext.Request.ContentType = "text/plain";
+            httpContext.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes("test"));
+
+            await middleware.Invoke(httpContext);
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("Protocol: HTTP/1.0"));
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("Method: GET"));
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("Scheme: http"));
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("Path: /foo"));
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("PathBase: /foo"));
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("QueryString: ?foo"));
+            Assert.Contains(TestSink.Writes, w => w.Message.Contains("Connection: keep-alive"));
+            Assert.DoesNotContain(TestSink.Writes, w => w.Message.Contains("Body: test"));
+        }
+
+        [Fact]
+        public async Task RequestLogsAllRequestInfo()
+        {
+            var options = CreateOptionsAccessor();
+            options.CurrentValue.LoggingFields = HttpLoggingFields.Request;
+            var middleware = new HttpLoggingMiddleware(
+                async c =>
+                {
+                    var arr = new byte[4096];
+                    while (true)
+                    {
+                        var res = await c.Request.Body.ReadAsync(arr);
+                        if (res == 0)
+                        {
+                            break;
+                        }
+                    }
+                },
+                options,
+                LoggerFactory);
+
+            var httpContext = new DefaultHttpContext();
```

Comment:
Helper method for returning a common httpcontext?

Reply:
I don't think it would simplify much. What do you have in mind?

Reply:
It removes all the duplicate lines in every test

Reply:
A lot of these differ by one or two lines.